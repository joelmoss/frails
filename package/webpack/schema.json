{
  "type": "object",
  "properties": {
    "devServer": {
      "anyOf": [
        {
          "description": "Enabled in development mode, but can be enabled/disabled completely by setting to `true`/`false`",
          "type": "boolean"
        },
        { "type": "object" }
      ]
    },
    "sideLoadViews": {
      "description": "Side load views with Javascript and CSS assets. Enabled by default (`true`).",
      "type": "boolean"
    }
  }
}
