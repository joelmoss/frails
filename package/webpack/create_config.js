const path = require('path')
const debug = require('debug')('frails')
const { default: merge } = require('webpack-merge')
const validate = require('schema-utils')
const webpack = require('webpack')
const WebpackAssetsManifest = require('webpack-assets-manifest')
const FixStyleOnlyEntriesPlugin = require('webpack-fix-style-only-entries')

const frailsConfig = require('../config')
const applyDevServer = require('./config/dev_server')
const applySideLoadAssets = require('./config/side_load_views')
const schema = require('./schema.json')

module.exports = (options = {}) => {
  validate(schema, options)

  // Make sure NODE_ENV env variable is set.
  process.env.NODE_ENV = frailsConfig.railsEnv === 'test' ? 'development' : frailsConfig.railsEnv

  const config = merge(
    // Default config.
    {
      mode: process.env.NODE_ENV,
      entry: {},

      // Merge with user-provided options.
      ...options
    },

    // Required config - this should be untouched.
    {
      output: {
        path: frailsConfig.absolutePublicPath,
        publicPath: `/${frailsConfig.publicOutputPath}/`
      },
      resolve: {
        alias: {
          assets: path.resolve(frailsConfig.appPath, 'assets')
        }
      },
      plugins: [
        new FixStyleOnlyEntriesPlugin(),
        new WebpackAssetsManifest({
          writeToDisk: true,
          entrypoints: true,
          sortManifest: false,
          publicPath: true
        })
      ]
    }
  )

  applyDevServer(config)
  applySideLoadAssets(config)

  debug(config)

  return config
}
