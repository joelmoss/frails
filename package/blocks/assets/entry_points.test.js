import { createConfig } from "@webpack-blocks/webpack";

import entryPoints from "./entry_points";

describe("assets/entryPoints", () => {
  test("config", () => {
    expect(createConfig(entryPoints())).toMatchSnapshot();
  });

  describe("entry()", () => {
    it("returns object of asset entry points", () => {
      const config = createConfig(entryPoints());

      expect(config.entry()).toMatchSnapshot();
    });
  });
});
