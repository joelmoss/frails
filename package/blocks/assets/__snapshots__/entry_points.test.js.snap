// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`assets/entryPoints config 1`] = `
Object {
  "entry": [Function],
  "module": Object {
    "rules": Array [],
  },
  "plugins": Array [],
  "resolve": Object {
    "extensions": Array [
      ".js",
      ".json",
    ],
  },
  "stats": Object {
    "children": false,
    "chunks": false,
    "modules": false,
    "reasons": false,
  },
}
`;

exports[`assets/entryPoints entry() returns object of asset entry points 1`] = `
Object {
  "assets/admin/index.entry": Array [
    "/Users/joelmoss/dev/frails/test/dummy/app/assets/admin/index.entry.js",
  ],
  "assets/app.entry": Array [
    "/Users/joelmoss/dev/frails/test/dummy/app/assets/app.entry.css",
  ],
  "assets/application.entry": Array [
    "/Users/joelmoss/dev/frails/test/dummy/app/assets/application.entry.js",
  ],
  "assets/blocks/assets/index.entry": Array [
    "/Users/joelmoss/dev/frails/test/dummy/app/assets/blocks/assets/index.entry.css",
    "/Users/joelmoss/dev/frails/test/dummy/app/assets/blocks/assets/index.entry.js",
  ],
  "assets/user.entry": Array [
    "/Users/joelmoss/dev/frails/test/dummy/app/assets/user.entry.css",
  ],
}
`;
